.github/workflows/ci.yml

```diff
@@ -138,4 +138,11 @@ jobs:
         with:
           token: ${{ secrets.CODECOV_TOKEN }}
           files: lcov.info
-          fail_ci_if_error: true
\ No newline at end of file
+          fail_ci_if_error: true
+  trigger-deployment:
```

Comment:
wouldn't this deployment run everytime a PR is raised? as the trigger condition in this `ci.yml` is
```
on:
  push:
    branches:
      - dev
    tags:
      - "[v]?[0-9]+.[0-9]+.[0-9]+*"
  pull_request:

```

Reply:
We can add this to the `trigger-tee-deployment` job:
```
    # only for dev branch and tags
    if: github.event_name == 'push' && (startsWith(github.ref, 'refs/tags/') || github.ref == 'refs/heads/dev')
```